junebug/stores.py

```diff
@@ -1,6 +1,17 @@
 from twisted.internet.defer import inlineCallbacks, returnValue
+from twisted.web import http
+
 from vumi.message import TransportUserMessage
 
+from junebug.error import JunebugError
+
+
+class MessageNotFound(JunebugError):
+    '''Raised when a message is not found.'''
+    name = 'MessageNotFound'
+    description = 'message not found'
+    code = http.NOT_FOUND
```

Comment:
It might also be a bit strange having the store raise a 400 error if it can't find something.